packages/codegen/src/generate.ts

```diff
@@ -792,6 +816,13 @@ export default class ApiGenerator {
     with a new name adding a number
   */
   getTrueEnum(schema: SchemaObject, propName: string) {
+    if (typeof schema == "boolean") {
+      // HACK: this should never be thrown, since the only `getTrueEnum` call is
+      // behind an `isTrueEnum` check, which returns false for boolean schemas.
+      throw new Error(
+        "cannot get enum from boolean schema. schema must be an object",
+      );
```

Comment:
Fair! Thanks for the clear comment! I'd personally not consider this a "HACK", it's just an additional guard that should never be invoked.